### PR TITLE
Add detail screen

### PR DESCRIPTION
**Pull request by gfontenot**
This actually isn't "finished", but I think it's worthwhile to get it
into master before working on the individual problems. I've created
issues for all of the known issues, but basically:
- [x] Properties from `TBEmployee` are implicitly unwrapped optionals. This
  can cause crashes if an employee doesn't have this info in their
  profile and we try to access it. Specifically, this is a problem for
  start dates and personal website urls.
- [ ] We're still showing cells even if the employee doesn't have
  information to use in that cell
- [x] We're being horribly inefficient with our use of `NSDateFormatter` in
  the employee view model. We should fix that.

### CODE REVIEW

File: roster/Classes/ViewControllers/EmployeesViewController.swift [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        TBTeamClient().fetchAllEmployeesWithCompletion { employees, _ in
-            self.viewModel = EmployeesViewModel(employees: employees as [TBEmployee])
+        TBTeamClient().fetchAllEmployeesWithCompletion { employees, error in
+            if employees == nil {
+                println(error)
+                return
+            }
```
Comment (gfontenot) [inline roster/Classes/ViewControllers/EmployeesViewController.swift]: Didn't mean to commit this. Removing.

**Comment by tonyd256** [comment]
:+1:

**Comment by gfontenot** [comment]
Updated the PR to fix some of the implicitly unwrapped optional stuff, as well as fixing the creation of website URLs

**Comment by gfontenot** [comment]
This PR now fixes #13, #12, #11, and #8

**Comment by gfontenot** [comment]
Oh, yeah, that implementation is much nicer.

File: roster/Classes/ViewControllers/EmployeesViewController.swift [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 class EmployeesViewController: UITableViewController {
-    var viewModel: EmployeesViewModel = EmployeesViewModel()
+    var viewModel: EmployeeListViewModel = EmployeeListViewModel()
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        TBTeamClient().fetchAllEmployeesWithCompletion { employees, _ in
-            self.viewModel = EmployeesViewModel(employees: employees as [TBEmployee])
+        TBTeamClient().fetchAllEmployeesWithCompletion { employees, error in
```
Comment (hyperspacemark) [inline roster/Classes/ViewControllers/EmployeesViewController.swift]: Where is `error` being used here?

Comment (gfontenot) [inline roster/Classes/ViewControllers/EmployeesViewController.swift]: Oh, it isn't. I think I needed it for debugging. I'll ignore it.

File: roster/Classes/ViewModels/EmployeeViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    func handleForUsername(name: String?) -> String {
+        let handle = usernameFormat <^> name
+
+        return handle ?? ""
+    }
+
+    func displayWebsiteForURL(url: NSURL?) -> String {
+        let string = websiteFormat <^> url?.host <*> url?.path
+
+        return string ?? ""
```
Comment (hyperspacemark) [inline roster/Classes/ViewModels/EmployeeViewModel.swift]: Could this basic function of taking an optional string and returning either the string or an empty string be abstracted further?

File: roster/Classes/ViewModels/EmployeeViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+private extension EmployeeViewModel {
+    func handleForUsername(name: String?) -> String {
+        let handle = usernameFormat <^> name
+
+        return handle ?? ""
+    }
+
+    func displayWebsiteForURL(url: NSURL?) -> String {
+        let string = websiteFormat <^> url?.host <*> url?.path
```
Comment (hyperspacemark) [inline roster/Classes/ViewModels/EmployeeViewModel.swift]: Just out of curiosity, is it possible to apply a tuple?

**Comment by hyperspacemark** [comment]
This is looking pretty good to me. Sprinkled a couple questions throughout.

File: roster/Classes/ViewModels/EmployeeViewModel.swift (hunk 4 above)
Comment (tonyd256) [inline roster/Classes/ViewModels/EmployeeViewModel.swift]: yes you could ... by then fmap / apply would check for the existence of the tuple which would always be there .. just the values inside could be .None ... in which case the function you are applying with would need to take 2 optional values ... so it wouldn't work in this scenario but yes you can use tuples as argument lists for functions.

**Pull request merged**